c_api.h

```diff
@@ -1,15 +1,96 @@
 #ifndef C_API_H
 #define C_API_H
 
-#include "c_secure_comm.h"
+#include <pthread.h>
+
+#define DIST_KEY_EXPIRATION_TIME_SIZE 6
+#define KEY_EXPIRATION_TIME_SIZE 6
+#define SESSION_KEY_ID_SIZE 8
+#define MAC_KEY_SIZE 32
+#define MAX_CIPHER_KEY_SIZE 32
+#define MAX_SESSION_KEY 10
+
+typedef struct {
+    unsigned char key_id[SESSION_KEY_ID_SIZE];
+    unsigned char abs_validity[KEY_EXPIRATION_TIME_SIZE];
+    unsigned char rel_validity[KEY_EXPIRATION_TIME_SIZE];
+    unsigned char mac_key[MAC_KEY_SIZE];
+    unsigned int mac_key_size;
+    unsigned char cipher_key[MAX_CIPHER_KEY_SIZE];
+    unsigned int cipher_key_size;
+    char enc_mode;
+    char no_hmac_mode;
+} session_key_t;
+
+typedef struct {
+    unsigned char mac_key[MAC_KEY_SIZE];
+    unsigned int mac_key_size;
+    unsigned char cipher_key[MAX_CIPHER_KEY_SIZE];
+    unsigned int cipher_key_size;
+    unsigned char abs_validity[DIST_KEY_EXPIRATION_TIME_SIZE];
+    char enc_mode;
+} distribution_key_t;
+
+typedef struct {
+    char name[32];
```

Comment:
Please use #define for constants.

Reply:
Fixed 02685c03a97ac4a72e99fff863ae7b67aef39143